scripts/deployment-test/cf-workers-esm.mjs

```diff
@@ -0,0 +1,128 @@
+import toml from "@iarna/toml";
+import PackageJson from "@npmcli/package-json";
+import { createApp } from "@remix-run/dev";
+import { sync as spawnSync } from "cross-spawn";
+import fse from "fs-extra";
+import fetch from "node-fetch";
+import path from "path";
+
+import {
+  addCypress,
+  CYPRESS_CONFIG,
+  CYPRESS_SOURCE_DIR,
+  getAppDirectory,
+  getAppName,
+  getSpawnOpts,
+  runCypress,
+  validatePackageVersions,
+} from "./_shared.mjs";
+
+let APP_NAME = getAppName("cf-workers-esm");
+let PROJECT_DIR = getAppDirectory(APP_NAME);
+let CYPRESS_DEV_URL = "http://localhost:8787";
+
+async function createNewApp() {
+  await createApp({
+    appTemplate: "cloudflare-workers-esm",
+    installDeps: false,
+    useTypeScript: true,
+    projectDir: PROJECT_DIR,
+  });
+}
+
+async function createAndDeployApp() {
+  // create a new remix app
+  await createNewApp();
+
+  // validate dependencies are available
+  let [valid, errors] = await validatePackageVersions(PROJECT_DIR);
+
+  if (!valid) {
+    console.error(errors);
+    process.exit(1);
+  }
+
+  let pkgJson = await PackageJson.load(PROJECT_DIR);
+  pkgJson.update({
+    devDependencies: {
+      ...pkgJson.content.devDependencies,
+      wrangler: "latest",
+    },
+  });
+
+  // add cypress to the project
+  await Promise.all([
+    fse.copy(CYPRESS_SOURCE_DIR, path.join(PROJECT_DIR, "cypress")),
+    fse.copy(CYPRESS_CONFIG, path.join(PROJECT_DIR, "cypress.json")),
+    addCypress(PROJECT_DIR, CYPRESS_DEV_URL),
+    pkgJson.save(),
+  ]);
+
+  let spawnOpts = getSpawnOpts(PROJECT_DIR, {
+    // these would usually be here by default, but I'd rather be explicit, so there is no spreading internally
+    CLOUDFLARE_API_TOKEN: process.env.CLOUDFLARE_API_TOKEN,
+    CLOUDFLARE_ACCOUNT_ID: process.env.CLOUDFLARE_ACCOUNT_ID,
+    CLOUDFLARE_GLOBAL_API_KEY: process.env.CLOUDFLARE_GLOBAL_API_KEY,
+    CLOUDFLARE_EMAIL: process.env.CLOUDFLARE_EMAIL,
+  });
+
+  // install deps
+  spawnSync("npm", ["install"], spawnOpts);
+  spawnSync("npm", ["run", "build"], spawnOpts);
+
+  // run cypress against the dev server
+  runCypress(PROJECT_DIR, true, CYPRESS_DEV_URL);
+
+  // we need to update the workers name
+  let wranglerTomlPath = path.join(PROJECT_DIR, "wrangler.toml");
+  let wranglerTomlContent = await fse.readFile(wranglerTomlPath);
+  let wranglerToml = toml.parse(wranglerTomlContent);
+  wranglerToml.name = APP_NAME;
+  await fse.writeFile(wranglerTomlPath, toml.stringify(wranglerToml));
+  let url = `https://${APP_NAME}.remixrun.workers.dev`;
+  console.log(`worker url: ${url}`);
```

Comment:
was this used for debugging and it should be removed? or it is actually here on purpose?

Reply:
This is just a copy of the same script for Service Workers. I assume it's intentional so that it shows up in the GitHub Actions logs when it runs.